public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -138,6 +139,9 @@ export interface DashboardSceneState extends SceneObjectState {
   panelsPerRow?: number;
   /** options pane */
   editPane: DashboardEditPane;
+
+  /** element panel mapping schema v2 */
+  elementPanelMapping?: ElementPanelMappingService;
```

Comment:
After having an overlook, I like @dprokop idea of delegating the responsibility of saving and assigning the keys to the serializer. 

Because each dashboard has a different map, having DashboardScene take care of it, makes sense; we probably don't need a singleton.